x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -203,12 +216,12 @@ func createCloudWatchEvents(getMetricDataOutput *cloudwatch.GetMetricDataOutput,
 		mapOfMetricSetFieldResults["instance.private.ip"] = *privateIP
```

Comment:
For all the values that you are assigning here you shouldn't need to apply an schema, you could set them directly in the resulting MetricSetFields (or RootFields).
Notice in the schema definition that you are assigning from a field to the same, i.e, with the following schema you are mapping from `instance.image.id` to `instance.image.id`. This shouldn't be needed.
```
		"instance": s.Object{
			"image": s.Object{
				"id": c.Str("instance.image.id", s.Optional),
                ...
```

Reply:
@jsoriano Yeah but that case, the schema in data.go will only show partial data structure not the full picture. Is that OK then? I somehow had the idea of we need to put all fields into data.go schema haha. If that's not the case, then yes, I can set some values directly in MetricSetFields and RootFields.